CIScripts/Test/TestConfigurationUtils.Tests.ps1

```diff
@@ -0,0 +1,48 @@
+
+. $PSScriptRoot\TestConfigurationUtils.ps1
+
+Describe "Select-ValidNetIPInterface unit tests" -Tags CI, Unit {
+        It "Both AddressFamily and SuffixOrigin match values in Select-ValidNetIPInterface" {
+            $TestResult = $MockedGetNetIPAddress | Select-ValidNetIPInterface
+            $TestResult | Should -eq $MockedGetNetIPAddress
+        }
+
+        It "AddressFamily isn't matching" {
+            $MockedGetNetIPAddress.AddressFamily = "IPv6"
+            $TestResult = $MockedGetNetIPAddress | Select-ValidNetIPInterface
+            $TestResult | Should BeNullOrEmpty
+        }
+
+        It "SuffixOrigin isn't matching" {
+            $MockedGetNetIPAddress.SuffixOrigin = "WellKnown", "Link", "Random"
+            $TestResult = $MockedGetNetIPAddress | Select-ValidNetIPInterface
+            $TestResult | Should BeNullOrEmpty
+        }
+
+        It "Both AddressFamily and SuffixOrigin do not match" {
+            $MockedGetNetIPAddress.AddressFamily = "IPv6"
+            $MockedGetNetIPAddress.SuffixOrigin = "WellKnown", "Link", "Random"
+            
+            $TestResult = $MockedGetNetIPAddress | Select-ValidNetIPInterface
+            $TestResult  | Should BeNullOrEmpty
+        }
+
+        It "Both AddressFamily and SuffixOrigin are empty strings" {
+            $MockedGetNetIPAddress.AddressFamily = ""
+            $MockedGetNetIPAddress.SuffixOrigin = ""
+
+            $TestResult = $MockedGetNetIPAddress | Select-ValidNetIPInterface
+            $TestResult | Should BeNullOrEmpty
+        }
+
```

Comment:
It would be also nice to have a test in which there are multiple objects passed to `Select-ValidNetIPInterface` (eg. the first one valid but second one not, and vice versa). I've recently seen a weird behaviour when function that uses `ValueFromPipeline` was looking only at the last value from list and I needed to use [`Process {}` block](https://docs.microsoft.com/en-us/powershell/module/microsoft.powershell.core/about/about_functions_advanced_methods?view=powershell-6#process).